### PR TITLE
collect cloud VM metadata

### PR DESCRIPTION
**Pull request by luke-zhu**
This change introduces code which polls for VM metadata every 5 minutes. Metadata collection can be enabled by
setting `--cloud_metadata TRUE`

I tested this in AWS and GCP using the test VMs. I tested on a single azure instance by running the curl command manually.

### CODE REVIEW

**Comment by arthur-observe** [comment]
Looks like the way this works is - 
We use https://docs.fluentbit.io/manual/pipeline/inputs/exec - exec input plugin to execute curl script that hits metadata endpoint for respective cloud

File: observe_configure_script.sh [hunk 1]
```diff
@@ -538,6 +562,27 @@ if [ "$appgroup" != UNSET ]; then
...
 fi
 
+metadata_buffer_size="8mb"
+metadata_interval_secs="300"
+sed -i "s/REPLACE_WITH_METADATA_BUFFER_SIZE/${metadata_buffer_size}/g" ./*
+sed -i "s/REPLACE_WITH_METADATA_INTERVAL/${metadata_interval_secs}/g" ./*
+
+metadata_command=":"
+if [[ "$cloud_metadata" == TRUE ]]; then
+  if [ $datacenter == "AWS" ]; then
```
Comment (arthur-observe) [inline observe_configure_script.sh:572]: Is there any way to auto detect the cloud we are on?  I worry about fragility of requiring the correct datacenter flag to make this work.

**Comment by arthur-observe** [comment]
@jlebaugh - we need to discuss how we want to test this and talk about any potential side effects

File: observe_configure_script.sh (hunk 1 above)
Comment (luke-zhu) [reply observe_configure_script.sh:572]: thanks, i think the easiest way is to run the curl command with a timeout and then check the bash status and http status. i'll try that out

**Comment by luke-zhu** [comment]
i think the tests are failing because the branch variable isn't passed in one of the 2 places (URL or config flag)

**Comment by luke-zhu** [comment]
### GCP
image

### Azure
image

**Comment by arthur-observe** [comment]
I'm working on fixing tests

**Comment by arthur-observe** [comment]
Pretty sure I deleted the datastream that token was for and blows this up

That will fix main but I also see that the tests are not passing branch version through to script for non-main - let me confirm - I think it's a simple fix

**Comment by luke-zhu** [comment]
thanks, i think the only issue i'm hitting is with certain resources already existing upon terraform apply

locally, i avoided the issue by introducing the name_format variable

about branch:  I added --branch_input to fabfile.py. It looks like this makes the .conf files get pulled from the branch instead of main.

**Comment by luke-zhu** [comment]
more photos in the Windows PR:
https://github.com/observeinc/windows-host-configuration-scripts/pull/8

**Review by arthur-observe — COMMENTED** [review]
I added concurrency to workflows to prevent competing runs

**Comment by arthur-observe** [comment]
> thanks, i think the only issue i'm hitting is with certain resources already existing upon terraform apply
> 
> locally, i avoided the issue by introducing the name_format variable
> 
> about branch: I added --branch_input to fabfile.py. It looks like this makes the .conf files get pulled from the branch instead of main.

Every once in a while these workflows leave terraform state broken - add commit to your branch to implement concurrency groups in workflow runs which will hopefully stop competing runs

**Comment by arthur-observe** [comment]
ok seems like telegraf is failing consistently across machines

**Review by arthur-observe — CHANGES_REQUESTED** [review]
Please rebase on main to uptake changes for telegraf repo key

File: test_code/python_scripts/fabfile.py [hunk 2]
```diff
@@ -498,78 +497,75 @@ def write_to_local_summary_results_file(sum_file, small_result, outPutTitleAppen
...
 
-        # info
-        logging.debug(
-            f"""
-            #######################################
-            {key}
+            # info
+            logging.debug(
+                f"""
+    `            #######################################
```
Comment (luke-zhu) [inline test_code/python_scripts/fabfile.py:533]: nit

**Review by arthur-observe — APPROVED** [review]

**Pull request merged**